lib/internal/modules/esm/loader.js

```diff
@@ -408,41 +467,94 @@ class ESMLoader {
 
     const namespaces = await PromiseAll(new SafeArrayIterator(jobs));
 
-    return wasArr ?
-      namespaces :
-      namespaces[0];
+    if (!wasArr) return namespaces[0];
+
+    for (let i = 0; i < count; i++) {
+      const namespace = ObjectCreate(null);
+      namespace.url = specifiers[i];
+      namespace.exports = namespaces[i];
+
+      namespaces[i] = namespace;
+    }
+
+    return namespaces;
   }
 
   /**
    * Provide source that is understood by one of Node's translators.
    *
-   * The internals of this WILL change when chaining is implemented,
-   * depending on the resolution/consensus from #36954
-   * @param {string} url The URL/path of the module to be loaded
+   * @param {URL['href']} url The URL/path of the module to be loaded
    * @param {object} context Metadata about the module
-   * @returns {object}
+   * @returns {{ format: ModuleFormat, source: ModuleSource }}
    */
   async load(url, context = {}) {
-    const defaultLoader = this.#loaders[0];
+    const loaders = this.#loaders;
+    let i = loaders.length - 1;
+    let {
+      fn: loader,
+      url: loaderFilePath,
+    } = loaders[i];
+    let chainFinished = i === 0;
+
+    function next(nextUrl, ctx = context) {
+      ({
+        fn: loader,
+        url: loaderFilePath,
+      } = loaders[--i]);
+
+      if (!i) chainFinished = true;
+
+      const hookErrIdentifier = `${loaderFilePath} "load"`;
+
+      if (typeof nextUrl !== 'string') { // non-strings can be coerced to a url string
+        throw new ERR_INVALID_ARG_TYPE(
+          `${hookErrIdentifier} next(url)`,
+          'a url string',
+          nextUrl,
+        );
+      }
 
-    const loader = this.#loaders.length === 1 ?
-      defaultLoader :
-      this.#loaders[1];
-    const loaded = await loader(url, context, defaultLoader);
+      try {
+        new URL(nextUrl);
+      } catch (err) { // eslint-disable-line no-unused-vars
```

Comment:
```suggestion
      } catch {
```

Reply:
Some check doesn't like that. I tried it first.

Reply:
Which check?

Reply:
🤦‍♂️ it was a typescript error from VS Code. updated